lib/Reader/reader.cpp

```diff
@@ -3486,17 +3486,22 @@ IRNode *ReaderBase::genericTokenToNode(CORINFO_RESOLVED_TOKEN *ResolvedToken,
   // No runtime lookup is required
   if (!Result.lookup.lookupKind.needsRuntimeLookup) {
     if (NeedResult) {
+      mdToken Token = ResolvedToken->token;
+      if ((Result.handleType == CORINFO_HANDLETYPE_CLASS) &&
+          (TypeFromToken(Token) != mdtClassHandle)) {
```

Comment:
Will add a comment:
      // If the handle really corresponds to a class and not a method, pass
      // mdtClassHandle to handleToIRNode so that a class name is used for the
      // name of the GlobalVariable created there instead of the method name.